### PR TITLE
EMMA-5807 add cuber cluster to terraform

### PR DESCRIPTION
**Pull request by tp4k**
https://emma-cloud.atlassian.net/browse/EMMA-5807

### CODE REVIEW

File: internal/emma/kubernetes_resource.go [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+		}
+		autoscalingConfig.ConfigurationPriorities = configurationPriorities
+
+		autoscalingConfigs = append(autoscalingConfigs, autoscalingConfig)
+	}
+
+	return autoscalingConfigs
+}
+
+func convertToConfigurationPriorities() {
```
Comment (DenisTyunin) [inline internal/emma/kubernetes_resource.go]: redundant func

File: internal/emma/kubernetes_resource.go [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+			RamGb:        int32(node.RamGb.ValueInt64()),
+			VolumeType:   node.VolumeType.ValueString(),
+			VolumeGb:     int32(node.VolumeGb.ValueInt64()),
+		})
+	}
+	kubernetes.WorkerNodes = workerNodes
+
+	autoscalingConfigs := convertToAutoScalingConfigs(data)
+	kubernetes.AutoscalingConfigs = autoscalingConfigs
+}
```
Comment (DenisTyunin) [inline internal/emma/kubernetes_resource.go:295]: Please add extra line between functions

File: internal/emma/kubernetes_resource.go [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+	tflog.Info(ctx, "Update kubernetes cluster")
+
+	var kubernetes emmaSdk.KubernetesUpdate
+	ConvertToKubernetesUpdateResourceRequest(ctx, planData, &kubernetes)
+	auth := context.WithValue(ctx, emmaSdk.ContextAccessToken, *r.token.AccessToken)
+	kubernetesGroup, response, err := r.apiClient.KubernetesClustersAPI.EditKubernetesCluster(auth, int32(stateData.Id.ValueInt64())).KubernetesUpdate(kubernetes).Execute()
+
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error",
+			fmt.Sprintf("Unable to create kubernetes cluster, got error: %s", tools.ExtractErrorMessage(response)))
```
Comment (DenisTyunin) [inline internal/emma/kubernetes_resource.go]: Please adjust error message

File: internal/emma/validation/kubernetes_resource_validation.go [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+	matches, _ := regexp.MatchString(`^([a-z](?:[0-9a-z-]{0,62}[0-9a-z]))$`, req.ConfigValue.ValueString())
+	if !matches {
+		resp.Diagnostics.AddError("Validation Error", "Action rejected: The name must be less than 64 characters, start with a lowercase letter, end with a lowercase alphanumeric, and use only lowercase alphanumerics and hyphens in between.")
+	}
+}
+
+type KubernetesResourceDomainName struct{}
+
+func (k KubernetesResourceDomainName) Description(ctx context.Context) string {
+	return "valid domain name"
```
Comment (DenisTyunin) [inline internal/emma/validation/kubernetes_resource_validation.go]: Please fix description

File: internal/emma/validation/kubernetes_resource_validation.go [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+}
+
+func (k KubernetesResourceDomainName) MarkdownDescription(ctx context.Context) string {
+	return "valid domain name"
```
Comment (DenisTyunin) [inline internal/emma/validation/kubernetes_resource_validation.go]: And here

File: internal/emma/validation/kubernetes_resource_validation.go [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+	return "The name must be less than 64 characters, start with a lowercase letter, end with a lowercase alphanumeric, and use only lowercase alphanumerics and hyphens in between."
+}
+
+func (k KubernetesResourceName) ValidateString(ctx context.Context, req validator.StringRequest, resp *validator.StringResponse) {
+	if req.ConfigValue.IsUnknown() || req.ConfigValue.IsNull() {
+		return
+	}
+	matches, _ := regexp.MatchString(`^([a-z](?:[0-9a-z-]{0,62}[0-9a-z]))$`, req.ConfigValue.ValueString())
+	if !matches {
+		resp.Diagnostics.AddError("Validation Error", "Action rejected: The name must be less than 64 characters, start with a lowercase letter, end with a lowercase alphanumeric, and use only lowercase alphanumerics and hyphens in between.")
```
Comment (DenisTyunin) [inline internal/emma/validation/kubernetes_resource_validation.go]: Will user understand what name must be valid?

File: internal/emma/validation/kubernetes_resource_validation.go [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+}
+
+func (w *WorkerNodeName) ValidateString(ctx context.Context, req validator.StringRequest, resp *validator.StringResponse) {
+	if req.ConfigValue.IsUnknown() || req.ConfigValue.IsNull() {
+		return
+	}
+	name := req.ConfigValue.ValueString()
+	matches, _ := regexp.MatchString(`^([a-z](?:[0-9a-z-]{0,61}[0-9a-z]))$`, name)
+	if !matches {
+		resp.Diagnostics.AddError("Validation Error", "Action rejected: The name must be less than 64 characters, start with a lowercase letter, end with a lowercase alphanumeric, and use only lowercase alphanumerics and hyphens in between.")
```
Comment (DenisTyunin) [inline internal/emma/validation/kubernetes_resource_validation.go]: And here will user understand what name must be valid?

File: internal/emma/validation/kubernetes_resource_validation.go (hunk 6 above)
Comment (tp4k) [reply internal/emma/validation/kubernetes_resource_validation.go]: Just copied from doc, fixed thanks

File: internal/emma/validation/kubernetes_resource_validation.go (hunk 4 above)
Comment (tp4k) [reply internal/emma/validation/kubernetes_resource_validation.go]: Done

**Review by DenisTyunin — APPROVED** [review]

**Pull request merged**